prody/proteins/interactions.py

```diff
@@ -1816,7 +1873,7 @@ def showInteractionsGraph(statistics, **kwargs):
               'HIS': 'H', 'HSD': 'H','HSE': 'H', 'LEU': 'L', 'ARG': 'R', 'TRP': 'W', 
               'ALA': 'A', 'VAL':'V', 'GLU': 'E', 'TYR': 'Y', 'MET': 'M'}
     
-    if len(statistics[0]) != 4:
+    if len(statistics[0]) != 5:
```

Comment:
probably you should check some other things here too

Reply:
also on line 1880, the error should probably say statistics not node_size

Reply:
I think this is the only place where I using statistics. I double-checked everything.

Reply:
I meant something besides length, which may not work for all types. Maybe do the type check first and then the length check

Reply:
> I meant something besides length, which may not work for all types. Maybe do the type check first and then the length check 

That should work for all types of interactions because statistics is constructed in the same way for all of them (mean distances, std), but I will double-check that.

Reply:
I double-checked, and it is working fine. We are using here statistics = calcStatisticsInteractions(interactions), and this function gives the same format of output for each type of interaction (see below): pair of residues, weight of interaction, distance, std for distance, and energy. So, each time, we have five elements.

[['THR55A-ASP58A', 1.75, 2.7707, 0.104578, '-2.41'],
 ['THR7A-LYS11A', 1.75, 2.869729, 0.125774, '-1.64'],
 ['ARG72A-GLN40A', 1.25, 2.92852, 0.182836, '-4.13'],
 ['LYS11A-GLU34A', 1.0, 2.57245, 0.038992, '-2.83'],
 ['LYS6A-LEU67A', 1.0, 2.7815, 0.053496, '-3.24'],
 ['GLU34A-ILE30A', 1.0, 2.801175, 0.125321, '-3.99'],
 ['GLU51A-TYR59A', 1.0, 2.828925, 0.105081, '-3.62'],
 ['GLU64A-GLN2A', 1.0, 2.853175, 0.08989, '-3.45'],
 ['ILE13A-VAL5A', 1.0, 2.766925, 0.150216, '-5.31'],
 ['SER65A-GLN62A', 1.0, 3.008825, 0.247378, '-3.06']...

Reply:
But users may not know that they have to calculate statistics first or they might want to use these functions with their own inputs, so we need errors that consider more situations. It’s best to check type first and length after